src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,47 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = props => {
+  const { todo, onClose } = props;
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(todo.userId)
+      .then(data => {
+        setUser(data);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
```

Comment:
Consider adding error handling for the `getUser` promise. This will help manage any errors that occur during the API call and prevent unhandled promise rejections.